Bug fixes for Orleans Bank Account sample.

Rossimac opened the pull request:
Balance now updates after Deposit or Withdraw operations.

Withdraw decrements now instead of incorrectly incrementing.

oising commented:
Removing the immutability of the Balance class sets a bad precedent for this sample. The idea of immutability is to prevent the class of bugs where grain A makes a change to a instance, then sends it to grain B -- on the same silo -- and then makes another change to the instance, which, because it's a reference/byref, will change the instance in grain B also.

Rossimac commented:
I fixed a couple of build errors in the commit. 

Also, I fixed the sample to allow the bank balance to actually persist the balance after a Withdrawal.
